pkg/util/cephcmds.go

```diff
@@ -0,0 +1,364 @@
+}
+
+// SetOMapKeyValue sets the given key and value into the provided Ceph omap name
+func SetOMapKeyValue(monitors, adminID, key, poolName, oMapName, oMapKey, keyValue string) error {
+	// Command: "rados <options> setomapval oMapName oMapKey keyValue"
+
+	_, _, err := execCommand(
+		"rados",
+		"-m", monitors,
+		"--id", adminID,
+		"--key="+key,
+		"-c", CephConfigPath,
+		"-p", poolName,
+		"setomapval", oMapName, oMapKey, keyValue)
+	if err != nil {
+		klog.Errorf("failed adding key (%s with value %s), to omap (%s) in "+
+			"pool (%s): (%v)", oMapKey, keyValue, oMapName, poolName, err)
+		return err
+	}
+
+	return nil
+}
+
+// ErrKeyNotFound is returned when requested key in omap is not found
+type ErrKeyNotFound struct {
+	keyName string
+	err     error
+}
+
+func (e ErrKeyNotFound) Error() string {
+	return e.err.Error()
+}
+
+// GetOMapValue gets the value for the given key from the named omap
+func GetOMapValue(monitors, adminID, key, poolName, oMapName, oMapKey string) (string, error) {
+	// Command: "rados <options> getomapval oMapName oMapKey <outfile>"
+	// No such key: replicapool/csi.volumes.directory.default/csi.volname
+	tmpFile, err := ioutil.TempFile("", "omap-get-")
+	if err != nil {
+		klog.Errorf("failed creating a temporary file for key contents")
+		return "", err
+	}
+	defer tmpFile.Close()
+	defer os.Remove(tmpFile.Name())
+
+	stdout, _, err := execCommand(
+		"rados",
+		"-m", monitors,
+		"--id", adminID,
+		"--key="+key,
+		"-c", CephConfigPath,
+		"-p", poolName,
+		"getomapval", oMapName, oMapKey, tmpFile.Name())
+	if err != nil {
+		// no logs, as attempting to check for key/value is done even on regular call sequences
+		if strings.Contains(string(stdout), "No such key: "+poolName+"/"+oMapName+"/"+oMapKey) {
+			return "", ErrKeyNotFound{poolName + "/" + oMapName + "/" + oMapKey, err}
+		}
+		return "", err
+	}
+
+	keyValue, err := ioutil.ReadAll(tmpFile)
+	return string(keyValue), err
+}
+
+// RemoveOMapKey removes the omap key from the given omap name
+func RemoveOMapKey(monitors, adminID, key, poolName, oMapName, oMapKey string) error {
+	// Command: "rados <options> rmomapkey oMapName oMapKey"
+
+	_, _, err := execCommand(
+		"rados",
+		"-m", monitors,
+		"--id", adminID,
+		"--key="+key,
+		"-c", CephConfigPath,
+		"-p", poolName,
+		"rmomapkey", oMapName, oMapKey)
+	if err != nil {
+		// NOTE: Missing omap key removal does not return an error
+		klog.Errorf("failed removing key (%s), from omap (%s) in "+
+			"pool (%s): (%v)", oMapKey, oMapName, poolName, err)
+		return err
+	}
+
+	return nil
+}
+
+// ErrOMapNotFound is returned when named omap is not found in rados
+type ErrOMapNotFound struct {
+	oMapName string
+	err      error
+}
+
+func (e ErrOMapNotFound) Error() string {
+	return e.err.Error()
+}
+
+// RemoveOMap removes the entire omap name passed in and returns ErrOMapNotFound is provided omap
+// is not found in rados
+func RemoveOMap(monitors, adminID, key, poolName, oMapName string) error {
```

Comment:
Nit: it's really just ```RemoveObject```

Reply:
ack! will address in a future update to the PR.